src/auth/Crypto.cc

```diff
@@ -420,6 +420,8 @@ int CryptoKey::_set_secret(int t, const bufferptr& s)
     if (error.length()) {
       return -EIO;
     }
+  } else {
+      return -EOPNOTSUPP;
```

Comment:
@ sorry, I am not familiar with ceph's test framework and no idea how to write a case of core dump, :) could you please help me to add it?

the crash can be reproduced simply(use a wrong keyring):
$ echo "BQAUlgtWoFePIxAAQ9YLzJSVgJX5V1lh5gyctg==" > /tmp/keyring
$ rados --keyfile /tmp/keyring -n client.admin lspools 

then a crash occurs, like following;
*** Caught signal (Segmentation fault) **
 in thread 7f95fe438700
 ceph version 9.2.0-920-g083c2e4 (083c2e42c663229ce505f74c40d8261ca530a79b)
 1: (ceph::BackTrace::BackTrace(int)+0x2d) [0x7f96093d4c21]
 2: (()+0xab31b) [0x7f96093d431b]

the strace information:
Program received signal SIGSEGV, Segmentation fault.
[Switching to Thread 0x7fffed10e700 (LWP 25051)]
0x00007ffff59896c6 in CryptoKey::encrypt (this=0x7fffed10d4f0, cct=0x555555829c30, in=..., out=..., error=0x7fffed10d440) at auth/cephx/../Crypto.h:110
110	    return ckh->encrypt(in, out, error);
(gdb) bt
#0  0x00007ffff59896c6 in CryptoKey::encrypt (this=0x7fffed10d4f0, cct=0x555555829c30, in=..., out=..., error=0x7fffed10d440)
    at auth/cephx/../Crypto.h:110

this is because during initialization, the function 
    CryptoKey::_set_secret()
        --> CryptoHandler::create() return NULL
but we do not catch it, which leds that CryptoKey::ckh not be assigned(NULL pointer). later when call encrypt() or decrypt(): 
        CryptoKey::encrypt()
            --> CryptoKey::ckh->encrypt() will led to a crash

this patch fixes it so that if user passes wrong secret, exit it gracefully instead of crash.